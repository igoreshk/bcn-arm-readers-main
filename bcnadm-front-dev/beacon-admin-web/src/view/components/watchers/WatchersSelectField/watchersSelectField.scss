.secondGroup .watcherSelectFormControl {
  display: flex;
  align-items: stretch;
  height: 32px;
  width: 250px;
  margin-left: 38px;
  background-color: #6b6a6c;
  border-radius: 20px;
  cursor: pointer;

  @media screen and (max-width: 625px) {
    width: 300px;
    margin-bottom: 20px;
    margin-left: 0;
  }

  .visitorsSelect {
    color: #ffffff;
  }

  .select {
    padding: 5px 15px;
    display: flex;
    align-items: center;
    font-size: 14px;
  }

  .selectIcon {
    color: #ffffff;
    margin-right: 10px;
  }
}

.visitorItem {
  .avatar {
    background: transparent;
    object-fit: cover;
  }

  .name {
    max-width: 160px;
    margin-left: 15px;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
