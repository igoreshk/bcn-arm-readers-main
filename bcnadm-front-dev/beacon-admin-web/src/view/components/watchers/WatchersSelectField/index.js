import React, { Component } from 'react';
import { FormControl, Select, MenuItem } from '@material-ui/core';
import PropTypes from 'prop-types';
import './watchersSelectField.scss';

class WatchersSelectField extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedWatchers: []
    };
  }

  handleChangeWatchers = (event) => {
    this.props.selectWatchersIds(event.target.value);
    this.setState({ selectedWatchers: event.target.value });
  };
  renderValue = (values) => {
    if (values.length === 0) {
      return <span className="placeholder">Select Watchers</span>;
    }

    if (values.length === 1) {
      return <span className="placeholder">{values[0].name}</span>;
    }

    if (values.length.toString().match(/(0|[2-9])1$/)) {
      return <span className="placeholder">Selected {values.length} watchers</span>;
    }

    if (values.length.toString().match(/^[2-4]$/) || values.length.toString().match(/(0|[2-9])[2-4]$/)) {
      return <span className="placeholder">Selected {values.length} watchers</span>;
    }

    return <span className="placeholder">Selected {values.length} watchers</span>;
  };

  render() {
    const { selectedWatchers } = this.state;
    const { watchers } = this.props;

    return (
      <FormControl className="watcherSelectFormControl">
        <Select
          multiple
          value={selectedWatchers}
          onChange={this.handleChangeWatchers}
          renderValue={this.renderValue}
          disableUnderline
          displayEmpty
          classes={{
            root: 'visitorsSelect',
            select: 'select',
            icon: 'selectIcon'
          }}
        >
          <MenuItem value="" disabled>
            <span>Select Watchers</span>
          </MenuItem>
          {watchers.map((watcher) => (
            <MenuItem name="watcher" key={watcher.entityId} value={watcher} className="visitorItem">
              <span className="name">{watcher.name}</span>
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    );
  }
}

WatchersSelectField.propTypes = {
  watchers: PropTypes.arrayOf(
    PropTypes.shape({
      entityId: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      visitorIds: PropTypes.arrayOf(PropTypes.string).isRequired
    }).isRequired
  ).isRequired,
  selectWatchersIds: PropTypes.func.isRequired
};

export default WatchersSelectField;
