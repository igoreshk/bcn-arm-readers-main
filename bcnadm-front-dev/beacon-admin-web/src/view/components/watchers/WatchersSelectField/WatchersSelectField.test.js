import React from 'react';
import { shallow } from 'enzyme';
import { Select } from '@material-ui/core';
import WatchersSelectField from './index';

const props = {
  watchers: [
    {
      entityId: '5fc8bf048dfe557a0c9a9191',
      id: '5fc8bf048dfe557a0c9a9191',
      visitorsId: ['12324']
    }
  ],
  selectWatchersIds: jest.fn()
};
const value = [{ id: 'test' }];
const wrapper = shallow(<WatchersSelectField {...props} />);

describe('WatchersSelectField', () => {
  it('Test for onChangeCall', () => {
    wrapper.find(Select).simulate('change', { target: { value } });
    expect(props.selectWatchersIds).toHaveBeenCalled();
  });

  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
});
