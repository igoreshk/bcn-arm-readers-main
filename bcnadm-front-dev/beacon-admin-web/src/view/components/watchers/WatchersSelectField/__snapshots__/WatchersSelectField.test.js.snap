// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchersSelectField Check matching snapshot 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="watcherSelectFormControl"
>
  <WithStyles(ForwardRef(Select))
    classes={
      Object {
        "icon": "selectIcon",
        "root": "visitorsSelect",
        "select": "select",
      }
    }
    disableUnderline={true}
    displayEmpty={true}
    multiple={true}
    onChange={[Function]}
    renderValue={[Function]}
    value={
      Array [
        Object {
          "id": "test",
        },
      ]
    }
  >
    <WithStyles(ForwardRef(MenuItem))
      disabled={true}
      value=""
    >
      <span>
        Select Watchers
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      className="visitorItem"
      key="5fc8bf048dfe557a0c9a9191"
      name="watcher"
      value={
        Object {
          "entityId": "5fc8bf048dfe557a0c9a9191",
          "id": "5fc8bf048dfe557a0c9a9191",
          "visitorsId": Array [
            "12324",
          ],
        }
      }
    >
      <span
        className="name"
      />
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
</WithStyles(ForwardRef(FormControl))>
`;
